### PR TITLE
Languages refactor

### PR DESCRIPTION
**Pull request by oferb**

### CODE REVIEW

File: public/embed/index.html [hunk 1]
```diff
@@ -39,10 +39,10 @@
     <link rel="stylesheet" href="../css/embed.css">
     <link rel="stylesheet" href="../css/styles.css" />
     <link rel="stylesheet" href="../css/map.css" />
-    <script src="../languages/He.js"></script>
-    <script src="../languages/En.js"></script>
-    <script src="../languages/Ar.js"></script>
     <script src="../languages/i18n.js"></script>
+    <script src="../languages/he.js"></script>
```
Comment (ShayMalchi) [inline public/embed/index.html:43]: is it after i18n on purpose?

File: public/index.html [hunk 2]
```diff
@@ -39,11 +39,11 @@
 
     <script defer src="https://code.jquery.com/jquery-3.4.1.min.js"
         integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
-    <script defer src="languages/He.js"></script>
-    <script defer src="languages/En.js"></script>
-    <script defer src="languages/Ar.js"></script>
-    <script defer src="languages/Ru.js"></script>
     <script defer src="languages/i18n.js"></script>
+    <script defer src="languages/he.js"></script>
```
Comment (ShayMalchi) [inline public/index.html:43]: same question

File: public/js/app.js [hunk 3]
```diff
@@ -132,6 +132,13 @@ const getQueryParam = (name) => {
   return urlParams.get(name);
 };
 
+const setQueryParam = (name, value) => {
+  let urlParams = new URLSearchParams(window.location.search);
+  urlParams.set(name, value);
+  urlParams.sort();
```
Comment (ShayMalchi) [inline public/js/app.js]: why is this necessary?

File: public/languages/i18n.js [hunk 4]
```diff
@@ -1,117 +1,67 @@
...
-const getDirection = () => {
-  const language = localStorage.getItem('language');
-  if (language === 'He' || language === 'Ar') {
-    return 'rtl';
-  } else {
-    return 'ltr';
-  }
+// Save language in local storage if not already there, update HTML
+const setLanguage = (language) => {
+  lang = language;
```
Comment (ShayMalchi) [inline public/languages/i18n.js]: this assignment is not clear. please use better naming i.e `currLang`  / `selectedLang`

**Review by ShayMalchi — APPROVED** [review]
LGTM, few questions

File: public/js/app.js (hunk 3 above)
Comment (oferb) [reply public/js/app.js]: So that our URL is always the same. It's better for CDN caching and analytics.

File: public/languages/i18n.js (hunk 4 above)
Comment (oferb) [reply public/languages/i18n.js]: Good idea

File: public/embed/index.html (hunk 1 above)
Comment (oferb) [reply public/embed/index.html:43]: Yup. Each language "registers itself" in a map variable that `i18n.js` declares.

File: public/js/app.js [hunk 5]
```diff
@@ -460,3 +448,11 @@ const removeURLParameter = (url, parameter) => {
     return url;
   }
 };
+
+const setMapReader = () => {
+  console.log("HEHEHEH" + getLanguage());
```
Comment (ShayMalchi) [inline public/js/app.js]: please remove
